packages/components/src/DataList/components/DataListHeader/DataListHeaderCheckbox.tsx

```diff
@@ -108,3 +103,45 @@ export function DataListHeaderCheckbox({ children }: DataListHeaderCheckbox) {
     return 0;
   }
 }
+
+function ColumnHeaderContent({
+  canSelectAll,
+  children,
+  hasAtLeastOneSelected,
+  selectedCount,
+  deselectText,
+  onSelect,
+}: {
+  readonly canSelectAll: boolean;
+  readonly children: ReactElement;
+  readonly hasAtLeastOneSelected: boolean;
+  readonly selectedCount: number;
+  readonly deselectText: string;
+  readonly onSelect:
+    | ((selected: DataListSelectedType<string | number>) => void)
+    | undefined;
```

Comment:
I'm wondering if there was a specific reason you chose union over optional property? I.e.
```suggestion
  readonly onSelect?: ((selected: DataListSelectedType<string | number>) => void);
```

Not necessarily requesting a change, just curious.

Reply:
fair question. to be honest Claude auto filled that and I hit Tab 😅 

I think there can be reason to do one approach over the other in certain situations, but in this case there was no real intent behind it.

Reply:
I also asked the AI to explain the benefits of using option 1 (your way) and option 2 (the suggested way above), and this is what it said:

> Practical Impact
>In most cases, Option 2 (optional property) is preferred because:
>- It's more idiomatic TypeScript
>- Allows cleaner object creation (can omit the property)
>- Better represents "this callback may or may not be provided"
>- More concise syntax
>
> Option 1 would be used when you specifically need to distinguish between "property exists but is undefined" vs "property doesn't exist" - which is rare in practice.
For callback properties like onSelect, Option 2 is the conventional choice.